src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateHelper.java

```diff
@@ -65,13 +71,45 @@ public final class CertificateHelper {
     public static final String PEM_BOUNDARY_CERTIFICATE = "CERTIFICATE";
     public static final String PEM_BOUNDARY_PUBLIC_KEY = "PUBLIC KEY";
     public static final String PEM_BOUNDARY_PRIVATE_KEY = "PRIVATE KEY";
+    private static final Map<ProviderType, String> providers = new HashMap<>();
+
 
     private CertificateHelper() {
     }
 
     static {
         // If not added "BC" is not recognized as the security provider
         Security.addProvider(new BouncyCastleProvider());
+        // Configure the default provider
+        providers.put(ProviderType.DEFAULT, "BC");
+    }
+
+    public enum ProviderType {
+        HSM, DEFAULT
+    }
+
+
+    /**
+     * Computes and returns the provider names from the registered providers in JCA that are required for issuing the
+     * certificates. When a user uses the PKCS11 provider it registers a provider by the name SunPKCS11-[name].
+     */
+    private static void computeProviders() {
```

Comment:
If this seems odd I would understand you, but this is the explanation - The PKCS11 provider component will create a SunPKCS11 provider, the way it works is that when we create our PKCS11 provider, it gets registered with a specific notation see documentation for the name config value [here](https://docs.oracle.com/javase/7/docs/technotes/guides/security/p11guide.html)

Given we don't know which name the user used on their configuration (it is on the PKCS11 Provider config) but we know that custom SunPKCS11 providers get named `SunPKCS11-[name]` this method tries to get the name of the registered provider so that then we can use it when issuing certificates